templates/github/.ci/scripts/check_release.py

```diff
@@ -39,17 +25,56 @@ def main():
         "'supported'. Defaults to 'supported', see `supported_release_branches` in "
         "`plugin_template.yml`.",
     )
-    opts = parser.parse_args()
+    return parser.parse_args()
+
+
+def template_config():
+    # Assume this script lies in .ci/scripts
+    path = Path(__file__).absolute().parent.parent.parent / "template_config.yml"
+    return yaml.safe_load(path.read_text())
 
+
+def current_version(repo, commitish):
+    try:
+        pyproject_toml = tomllib.loads(repo.git.show(f"{commitish}:pyproject.toml"))
+        current_version = pyproject_toml["project"]["version"]
+    except Exception:
+        current_version = repo.git.grep(
+            "current_version", commitish, "--", ".bumpversion.cfg"
+        ).split("=")[-1]
+    return Version(current_version)
+
+
+def check_pyproject_dependencies(repo, from_commit, to_commit):
+    try:
+        old_pyproject = tomllib.load(repo.show("{from_commit}:pyproject.toml"))
+        old_dependencies = set(old_pyproject["project"]["dependencies"])
+        new_pyproject = tomllib.load(repo.show("{to_commit}:pyproject.toml"))
+        new_dependencies = set(new_pyproject["project"]["dependencies"])
+        if old_dependencies != new_dependencies:
+            return ["dependencies"]
+        else:
+            return []
+    except Exception:
+        # Gathering more details failed.
+        return ["pyproject.toml changed somehow (PLEASE check if dependencies are affected)."]
```

Comment:
Maybe adding some info about the exception could be useful to know why gathering info failed